src/java/voldemort/store/readonly/ReadOnlyStorageEngine.java

```diff
@@ -176,6 +178,27 @@ public String getChunkIdToNumChunks() {
         return builder.toString();
     }
 
+    public void reSyncFileSystem() {
+        fileModificationLock.readLock().lock();
+        File latestSymLink = new File(latestDirSymLinkPath);
+        boolean needToResync = !Utils.symlinkExists(ReadOnlyUtils.getCurrentVersion(storeDir), latestSymLink);
+        fileModificationLock.readLock().unlock();
+
+        if (needToResync) {
+            logger.info("The internal state is inconsistent with the filesystem's state. Resyncing.");
+            File newLatestVersion = null;
+            try {
+                newLatestVersion = latestSymLink.getCanonicalFile();
```

Comment:
We just got this in `symlinkExists()` under the lock.  Could the `symlinkExists()` API be adjusted to make this more efficient and safe?